src/jit/importer.cpp

```diff
@@ -3927,6 +3927,44 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Buffers_Binary_BinaryPrimitives_ReverseEndianness:
+            {
+                assert(sig->numArgs == 1);
+
+                // We expect the return type of the ReverseEndianness routine to match the type of the
+                // one and only argument to the method. We use a special instruction for 16-bit
+                // BSWAPs since on x86 processors this is implemented as ROR value, 8. Additionally,
+                // we only allow BSWAPs with 64-bit arguments on 64-bit architectures.
```

Comment:
This last line of the comment confused me a bit; had to read it a few times and look at the code to get it parsed properly.

Perhaps "BSWAPs with 64-bit arguments are only supported on 64-bit architectures"?